test/IntegrationTests.t.sol

```diff
@@ -41,141 +44,421 @@ contract IntegrationTests is Test {
+            _checkUserActionsUnfrozen(
+                asset,
+                supplyAmount,
+                withdrawAmount,
+                borrowAmount,
+                repayAmount
+            );
 
-        vm.stopPrank();
+            vm.revertTo(snapshot);
+        }
 
-        // 4. Simulate spell after freeze, unfreezing market
-        vm.prank(SPARK_PROXY);
-        poolConfig.setReserveFreeze(WETH, false);
+        assertEq(untestedReserves.length, 1);
+        assertEq(untestedReserves[0],     0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599);  // WBTC
+    }
 
-        assertEq(_isFrozen(WETH), false);
+}
 
-        // 5. Check user actions after unfreeze
+contract FreezeAllAssetsSpellTest is IntegrationTestsBase {
 
-        vm.startPrank(sparkUser);
+    using SafeERC20 for IERC20;
+
+    address[] public untestedReserves;
 
-        // User can supply, borrow, repay, and withdraw
-        pool.supply(WETH, 10 ether, sparkUser, 0);
-        pool.borrow(WETH, 1 ether, 2, 0, sparkUser);
-        pool.repay(WETH, 1 ether, 2, sparkUser);
-        pool.withdraw(WETH, 1 ether, sparkUser);
+    function setUp() public override {
+        super.setUp();
+        vm.prank(SPARK_PROXY);
+        aclManager.addRiskAdmin(address(freezer));
     }
 
-    function _vote(address spell) internal {
-        uint256 amount = 1_000_000 ether;
+    function test_freezeAllAssetsSpell() external {
+        address[] memory reserves = pool.getReservesList();
 
-        deal(MKR, mkrWhale, amount);
+        assertEq(reserves.length, 9);
 
-        vm.startPrank(mkrWhale);
-        IERC20(MKR).approve(AUTHORITY, amount);
-        authority.lock(amount);
+        uint256 supplyAmount   = 1_000;
+        uint256 withdrawAmount = 1;
+        uint256 borrowAmount   = 2;
+        uint256 repayAmount    = 1;
 
-        address[] memory slate = new address[](1);
-        slate[0] = spell;
-        authority.vote(slate);
+        address freezeAllAssetsSpell = address(new FreezeAllAssetsSpell(address(freezer)));
 
-        vm.roll(block.number + 1);
+        // Setup spell
+        _vote(freezeAllAssetsSpell);
 
-        authority.lift(spell);
+        // Check that protocol is working as expected before spell for each asset
+        for (uint256 i = 0; i < reserves.length; i++) {
+            address asset    = reserves[i];
+            uint256 decimals = IERC20(asset).decimals();
 
-        vm.stopPrank();
+            // If the asset is already frozen on mainnet, skip the test
+            if (_isFrozen(asset)) {
+                untestedReserves.push(asset);
+                continue;
+            }
 
-        assertTrue(authority.hat() == spell);
-    }
+            // Max out supply caps for this asset so that they aren't hit during test
+            vm.startPrank(SPARK_PROXY);
+            poolConfig.setSupplyCap(asset, 68_719_476_735);  // MAX_SUPPLY_CAP
+            poolConfig.setBorrowCap(asset, 68_719_476_735);  // MAX_BORROW_CAP
```

Comment:
These can just be set to `0` to disable.

Reply:
Updated